app/gui/src/presenter.rs

```diff
@@ -113,23 +129,75 @@ impl Model {
         if let Ok(template) =
             template.map(double_representation::name::project::Template::from_text).transpose()
         {
-            crate::executor::global::spawn(async move {
+            crate::executor::global::spawn(with_progress_indicator(|| async move {
                 if let Ok(managing_api) = controller.manage_projects() {
-                    if let Err(err) = managing_api.create_new_project(template.clone()).await {
+                    if let Err(err) = managing_api.create_new_project(None, template.clone()).await
+                    {
                         if let Some(template) = template {
                             error!("Could not create new project from template {template}: {err}.");
                         } else {
                             error!("Could not create new project: {err}.");
                         }
                     }
                 } else {
-                    warn!("Project creation failed: no ProjectManagingAPI available.");
+                    warn!("Project Manager API not available, cannot create project.");
                 }
-            })
+            }))
         } else if let Some(template) = template {
             error!("Invalid project template name: {template}");
         };
     }
+
+    /// Open a project by name or ID. If no project with the given name exists, it will be created.
+    #[profile(Task)]
+    fn open_or_create_project(&self, project: ProjectToOpen) {
+        let controller = self.controller.clone_ref();
+        crate::executor::global::spawn(with_progress_indicator(|| async move {
+            if let Ok(managing_api) = controller.manage_projects() {
+                if let Err(error) = managing_api.open_or_create_project(project).await {
+                    error!("Cannot open or create project. {error}");
+                }
+            } else {
+                warn!("Project Manager API not available, cannot open or create project.");
+            }
+        }));
+    }
+}
+
+/// Show a full-screen spinner for the exact duration of the specified function.
+async fn with_progress_indicator<F, T>(f: F)
+where
+    F: FnOnce() -> T,
+    T: Future<Output = ()>, {
+    // TODO: Use a safer variant of getting the JS app.
```

Comment:
This TODO does not have initials and does not explain why it is unsafe. Every TODO/FIXME should always describe why it is bad and what is the right way of doing it, preferably linking to a ticket about it. I'm OK with not having the ticket, but the description should be understandable by any code reader after longer period of time. This tells about "safer variant" but does not describe what is the safer variant nor why this is unsafe.